application/src/main/java/org/opentripplanner/service/vehiclerental/model/VehicleRentalVehicle.java

```diff
@@ -2,74 +2,171 @@
+import java.util.Objects;
 import java.util.Set;
+import java.util.function.Consumer;
+import javax.annotation.Nullable;
 import org.opentripplanner.framework.i18n.I18NString;
 import org.opentripplanner.street.model.RentalFormFactor;
 import org.opentripplanner.transit.model.framework.FeedScopedId;
+import org.opentripplanner.utils.tostring.ToStringBuilder;
 
 /**
  * Implements the {@link VehicleRentalPlace} class which contains Javadoc.
+ * <p>
+ * THIS CLASS IS IMMUTABLE AND THREAD-SAFE.
  */
-public class VehicleRentalVehicle implements VehicleRentalPlace {
-
-  public FeedScopedId id;
-  public I18NString name;
-  public double longitude;
-  public double latitude;
-
-  public VehicleRentalSystem system;
-  public RentalVehicleType vehicleType;
-  public VehicleRentalStationUris rentalUris;
-  public boolean isReserved = false;
-  public boolean isDisabled = false;
-  public Instant lastReported;
-  public VehicleRentalStation station;
-  public String pricingPlanId;
-  public RentalVehicleFuel fuel;
-  public OffsetDateTime availableUntil;
-
-  @Override
-  public FeedScopedId getId() {
-    return id;
+public final class VehicleRentalVehicle implements VehicleRentalPlace {
+
+  public static final VehicleRentalVehicle DEFAULT = new VehicleRentalVehicle();
+
+  private final FeedScopedId id;
+  private final I18NString name;
+  private final double longitude;
+  private final double latitude;
+
+  private final VehicleRentalSystem system;
+  private final RentalVehicleType vehicleType;
+  private final VehicleRentalStationUris rentalUris;
+  private final boolean isReserved;
+  private final boolean isDisabled;
+  private final Instant lastReported;
+  private final VehicleRentalStation station;
+  private final String pricingPlanId;
+  private final RentalVehicleFuel fuel;
+  private final OffsetDateTime availableUntil;
+
+  public VehicleRentalVehicle() {
+    this.id = null;
+    this.name = null;
+    this.longitude = 0.0;
+    this.latitude = 0.0;
+    this.system = null;
+    this.vehicleType = null;
+    this.rentalUris = null;
+    this.isReserved = false;
+    this.isDisabled = false;
+    this.lastReported = null;
+    this.station = null;
+    this.pricingPlanId = null;
+    this.fuel = null;
+    this.availableUntil = null;
   }
 
-  @Override
-  public String getStationId() {
-    return getId().getId();
+  private VehicleRentalVehicle(Builder builder) {
+    this.id = builder.id;
+    this.name = builder.name;
+    this.longitude = builder.longitude;
+    this.latitude = builder.latitude;
+    this.system = builder.system;
+    this.vehicleType = builder.vehicleType;
+    this.rentalUris = builder.rentalUris;
+    this.isReserved = builder.isReserved;
+    this.isDisabled = builder.isDisabled;
+    this.lastReported = builder.lastReported;
+    this.station = builder.station;
+    this.pricingPlanId = builder.pricingPlanId;
+    this.fuel = builder.fuel;
+    this.availableUntil = builder.availableUntil;
```

Comment:
Some of these fields are required, should we enforce that here?

Reply:
I think at least Transmodel API has some of these fields set as required (although I'm not sure if all of those should be). It's a bit ugly if fetching rental vehicles fails because some of these fields are null.

Reply:
The only field that is required in the transmodel api RentalVehicle type is `vehicleType`, which, incidentally is *not* required by the spec. So not sure where that leaves us.

Reply:
`Objects#requireNonNull` for the id at least. Perhaps `DoubleUtils#requireInRange` for the coordinates but those are not strictly required, but do we import vehicles that are only part of a station? Also, what does happen if the check fails? Is only the individual vehicle not imported? I think that probably should be the behaviour.